### ISSUE TITLE
Custom univariate sampler not working as expected

### ISSUE THREAD

**Issue by nluetts** [JuliaStats/Distributions.jl#1132]
Implementing a univariate sampler as described [in the documentation](https://juliastats.org/Distributions.jl/v0.23.4/extends/#Univariate-Sampler-1), i.e. just implementing the function

```julia
function rand(s::Spl)
    # ... generate a single sample from s
end
```
is not working as expected. Specifically, the documentation says:

> The package already implements a vectorized version of rand! and rand that repeatedly calls the he scalar version to generate multiple samples.

So after implementing the function above, I was expecting generating several samples at once (calling `rand(mysampler, 10000)`) to work, but it throws an error. However, it works fine if the function `rand(rng::AbstractRNG, s::Spl)` is implemented.

For a full code example, see [this discussion on julia discourse](https://discourse.julialang.org/t/implementing-a-custom-sampler-in-distributions-jl/41066).

Is the documentation regarding custom samplers perhaps outdated?

**Comment by devmotion** [comment]
Would be fixed by https://github.com/JuliaStats/Distributions.jl/pull/1114.

**Comment by IoachimusRoderici** [comment]
I think the docs have been fixed as of 0.25 [docs](https://juliastats.org/Distributions.jl/v0.25/extends/#Univariate-Sampler).

Should this issue be closed?

**Comment by devmotion** [comment]
Indeed, this issue seems to be fixed 🙂
